src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java

```diff
@@ -256,4 +256,45 @@ private TableColumn<BibEntryTableViewModel, List<LinkedFile>> createExtraFileCol
     private TableColumn<BibEntryTableViewModel, String> createLibraryColumn(MainTableColumnModel columnModel) {
         return new LibraryColumn(columnModel);
     }
+
+    /**
+     * Create any single column
+     */
+    public TableColumn<BibEntryTableViewModel, ?> createColumn(MainTableColumnModel columnModel) {
+        switch (columnModel.getType()) {
+            case INDEX:
+                return createIndexColumn(columnModel);
+            case GROUPS:
+                return createGroupColumn(columnModel);
+            case FILES:
+                return createFilesColumn(columnModel);
+            case LINKED_IDENTIFIER:
+                return createIdentifierColumn(columnModel);
+            case LIBRARY_NAME:
+                return createLibraryColumn(columnModel);
+            case EXTRAFILE:
+                if (columnModel.getQualifier().isBlank()) {
+                    return createExtraFileColumn(columnModel);
+                }
+                break;
+            case SPECIALFIELD:
+                if (!columnModel.getQualifier().isBlank()) {
+                    Field field = FieldFactory.parseField(columnModel.getQualifier());
+                    if (field instanceof SpecialField) {
+                        return createSpecialFieldColumn(columnModel);
+                    } else {
+                        LOGGER.warn("Special field type '{}' is unknown. Using normal column type.", columnModel.getQualifier());
+                        return createFieldColumn(columnModel);
+                    }
+                }
+                break;
+            default:
+            case NORMALFIELD:
+                if (!columnModel.getQualifier().isBlank()) {
+                    return createFieldColumn(columnModel);
+                }
+                break;
+        }
+        return null;
+    }
```

Comment:
If you introduce this method, you need to adjust the constructor of the MainTableColumnFactory to reduce code duplication